File: c#/program29.cs
using System;

interface ArrayX
{
    void Accept();
    void Display();
}

struct MyArray : ArrayX
{
    public int[] Arr;

    public MyArray(int size)
    {
        this.Arr = new int[size];
    }

    public void Accept()
    {
        Console.WriteLine("Enter the elements of the array :");

        for(int iCnt = 0; iCnt  < Arr.Length; iCnt++)
        {
            this.Arr[iCnt] = int.Parse(Console.ReadLine());
        }
    }
    public void Display()
    {
        Console.WriteLine("Elements of the array are :");

        foreach(int iCnt in Arr)
        {
            Console.Write(iCnt+"\t");
        }
        Console.WriteLine();
    }
    public int CountOdd()
    {
        int count = 0;

        foreach(int iCnt in Arr)
        {
            if((iCnt%2) != 0)
            {
                count++;
            }
        }
        return count;
    }
}

struct MainStruct
{
    public static void Main()
    {
        int size = 0,ret = 0;

        Console.WriteLine("Enter the size of the array :");
        size = int.Parse(Console.ReadLine());

        MyArray obj = new MyArray(size);
        obj.Accept();
        obj.Display();

        ret = obj.CountOdd();
        Console.WriteLine("Count of the odd elements is : "+ret);
    }
}

File: java/program29.java
import java.util.*;

class ArrayX
{
    protected int Arr[];
    
    public ArrayX(int iSize)
    {
        Arr = new int[iSize];
    }

    protected void Accept()
    {
        int iCnt = 0;

        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter elements of array");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    protected void Display()
    {
        System.out.println("Elements of the array are : ");

        for(int iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.print(Arr[iCnt]+"\t");
        }
        System.out.println();
    }
}

class MyArray extends ArrayX
{
    public MyArray(int iSize)
    {
        super(iSize);
    }

    public int CountOdd()
    {
        int iCnt = 0;
        int iCount = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if((Arr[iCnt]%2) != 0)
            {
                iCount++;
            }
        }
        return iCount;
    }
}

class program29
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);
        int iValue = 0;

        System.out.println("Enter number of elements ");
        iValue = sobj.nextInt();

        MyArray obj = new MyArray(iValue);
        obj.Accept();
        obj.Display();
        
        System.out.println("Count of the Odd elements are : "+obj.CountOdd());
    }
}